### DISCUSSION TITLE
Custom name of Span when using Langchain handler instead of RunnableSequence

### DISCUSSION THREAD

**Discussion by nathan-vo810** [langfuse/langfuse#2009] [Support] [answered]
Example when running the following code:

```
@observe()  # decorator to automatically log function as trace to Langfuse
def main():
    # get the langchain handler for the current trace
    langfuse_handler = langfuse_context.get_current_langchain_handler()
    # invoke chain with langfuse handler
    chain.invoke({"person": "Alan Turing"},
                 config={"callbacks": [langfuse_handler]})
 
main()
```

The displayed name for the span inside `main` trace will be `RunnableSequence`.

<img width="364" alt="Screenshot 2024-05-08 at 14 21 30" src="https://github.com/langfuse/langfuse/assets/14048514/91fa3d34-d8dc-472b-8989-bff308352db3">

Is it possible to give another name to it?

**Comment by marcklingen** [comment]
Does his help? https://langfuse.com/docs/integrations/langchain/tracing#customizing-the-names-of-trace-objects

**Reply by nathan-vo810** [reply]
Unfortunately, that can only update the trace's name, which is the function name - `main`, but not the span's name (RunnableSequence). 

I would like to update the name of the span which is created from running the `chain.invoke` from RunnableSequence to something else. The reason is in 1 function I call multiple `chain.invoke` and all of the have the same RunnableSequence name which is not very meaningful.

**Reply by nathan-vo810** [reply]
One note is when passing `langfuse_handler` to langchain built-in chain, the span can take the name of it. For example: `ConversationalRetrievalChain` or `StuffDocumentChain`.

The RunnableSequence appear when building chain: 
`chain = prompt | llm `

**Reply by marcklingen** [reply]
This is how langfuse extracts the name for each trace and its observations. Feedback/improvements appreciated if there is a way to help you customize names in a more meaningful way.

https://github.com/langfuse/langfuse-python/blob/d71a3a2b347ad899d566eae0a9d1620e22c97eb8/langfuse/callback/langchain.py#L124

**Reply by nathan-vo810** [reply, marked as answer]
Thanks for the reference. I figured it out! 

Simply assign `chain.name="New name"` solved the problem.

**Reply by marcklingen** [reply]
awesome!
